Add pre-flight auth check to claude-wrapper.sh

## Context

During shepherd runs for issue #2270, agent sessions launched with `CLAUDE_CONFIG_DIR` override showed "Not logged in · Run /login" in the Claude Code UI. The sessions silently produced no useful output, and the shepherd had to exhaust all retry attempts before failing.

The existing pre-flight checks in `claude-wrapper.sh` verify:
- Claude CLI found
- API endpoint reachable (curl)
- MCP server health

But they do **not** verify authentication status.

## Proposal

Add a pre-flight auth check using `claude auth status --json` (with `CLAUDE_CONFIG_DIR` set) that:
1. Verifies `loggedIn: true` before launching the session
2. On failure, logs a clear error message identifying the auth issue
3. Exits early with a specific exit code so the shepherd can report `NEEDS_INTERVENTION` immediately rather than wasting 4 retry cycles

## Expected behavior

```
[INFO] Running pre-flight checks...
[INFO] Claude CLI found: /opt/homebrew/bin/claude
[INFO] API endpoint reachable (curl)
[ERROR] Authentication check failed: not logged in (CLAUDE_CONFIG_DIR=/path/to/config)
[ERROR] Run: CLAUDE_CONFIG_DIR=/path/to/config claude auth login
```

## Files

- `.loom/scripts/claude-wrapper.sh` — add auth check after existing pre-flight checks

---
*Filed from shepherd reflection on issue #2270.*

--- a/defaults/scripts/claude-wrapper.sh
+++ b/defaults/scripts/claude-wrapper.sh
@@ -303,6 +303,50 @@ check_cli_available() {
     return 0
 }
 
+# Pre-flight check: verify authentication status
+# Uses `claude auth status --json` to confirm the CLI is logged in.
+# When CLAUDE_CONFIG_DIR is set, passes it through so the check uses
+# the same config the session will use.
+check_auth_status() {
+    local auth_output
+    local auth_exit_code
+
+    # Unset CLAUDECODE to avoid nested-session guard when running inside
+    # a Claude Code session (e.g., during testing or shepherd-spawned builds).
+    auth_output=$(CLAUDECODE='' claude auth status --json 2>&1) || auth_exit_code=$?
+    auth_exit_code="${auth_exit_code:-0}"
+
+    if [[ "${auth_exit_code}" -ne 0 ]]; then
+        log_error "Authentication check command failed (exit ${auth_exit_code})"
+        log_error "Output: ${auth_output}"
+        if [[ -n "${CLAUDE_CONFIG_DIR:-}" ]]; then
+            log_error "CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR}"
+            log_error "Run: CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR} claude auth login"
+        else
+            log_error "Run: claude auth login"
+        fi
+        return 1
+    fi
+
+    # Parse the loggedIn field from JSON output
+    local logged_in
+    logged_in=$(echo "${auth_output}" | python3 -c "import json,sys; print(json.load(sys.stdin).get('loggedIn', False))" 2>/dev/null || echo "")
+
+    if [[ "${logged_in}" != "True" ]]; then
+        log_error "Authentication check failed: not logged in"
+        if [[ -n "${CLAUDE_CONFIG_DIR:-}" ]]; then
+            log_error "CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR}"
+            log_error "Run: CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR} claude auth login"
+        else
+            log_error "Run: claude auth login"
+        fi
+        return 1
+    fi
+
+    log_info "Authentication check passed (logged in)"
+    return 0
+}
+
 # Pre-flight check: verify API is reachable
 # Uses a lightweight HEAD request to api.anthropic.com
 check_api_reachable() {
@@ -622,6 +666,11 @@ run_preflight_checks() {
     fi
 
     check_api_reachable  # Non-fatal, just logs
+
+    if ! check_auth_status; then
+        log_error "Authentication pre-flight check failed"
+        return 1
+    fi
 
     if ! check_mcp_server; then
         log_error "MCP server pre-flight check failed"

